### PR TITLE
Base DB model updates

### PR DESCRIPTION
**Pull request by ddavidov**
- find() method added
- specification to entity class on retrieving the record

### CODE REVIEW

File: zoolanders/Framework/Service/Date.php [hunk 1]
```diff
@@ -17,13 +17,20 @@
...
 
+use Zoolanders\Framework\Container\Container;
 use Zoolanders\Framework\Utils\IsString;
 
 class Date
 {
 
     use IsString;
 
+    protected $container;
```
Comment (Skullbock) [inline zoolanders/Framework/Service/Date.php]: Don't do this... the entire point of my code is to remove the container injection in the constructor...

File: zoolanders/Framework/Controller/Controller.php [hunk 2]
```diff
@@ -96,7 +97,8 @@ public function getView($name = null, $config = array())
         $viewName = !empty($name) ? $name : $this->getName();
         $config['name'] = $viewName;
 
-        $view = $this->container->factory->view($this->input, $config);
```
Comment (Skullbock) [inline zoolanders/Framework/Controller/Controller.php]: We don't need this anymore. The getView method is not required, when a controller task needs one, it should just request it via either the controller constructor or, better, in the method signature

File: zoolanders/Framework/Model/Database.php [hunk 3]
```diff
@@ -393,4 +405,21 @@ protected function setupOperatorAndValue(&$operator, &$value)
...
+     * @param $key
+     *
+     * @return mixed
+     */
+    public function find($key){
+
+        $this->where($this->primary_key, '=', $key);
+        $this->buildQuery();
+
+        $record = $this->container->db->queryObject($this->query, $this->entity_class);
```
Comment (Skullbock) [inline zoolanders/Framework/Model/Database.php]: this should be $this->db->

File: zoolanders/Framework/Controller/Controller.php (hunk 2 above)
Comment (ddavidov) [inline zoolanders/Framework/Controller/Controller.php]: Will remove getView method

File: zoolanders/Framework/Service/Date.php (hunk 1 above)
Comment (ddavidov) [inline zoolanders/Framework/Service/Date.php]: Updated

File: zoolanders/Framework/Model/Database.php (hunk 3 above)
Comment (ddavidov) [inline zoolanders/Framework/Model/Database.php]: Updated

**Pull request merged**